code/game/objects/effects/contraband.dm

```diff
@@ -21,16 +21,25 @@
 	// posters store what name and description they would like their
 	// rolled up form to take.
 	if(poster_structure)
+		if(QDELETED(poster_structure))
+			stack_trace("A poster was initialized with a qdeleted poster_structure, something's gone wrong")
+			return INITIALIZE_HINT_QDEL
 		name = poster_structure.poster_item_name
 		desc = poster_structure.poster_item_desc
 		icon_state = poster_structure.poster_item_icon_state
 
 		name = "[name] - [poster_structure.original_name]"
+		//If the poster structure is being deleted something has gone wrong, kill yourself off too
+		RegisterSignal(poster_structure, COMSIG_PARENT_QDELETING, .proc/react_to_deletion)
```

Comment:
Does the parent structure have a reference to the poster?

Reply:
Quick search says no